backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -794,10 +794,28 @@ func (a *applicationEventLoopRunner_Action) applicationEventRunner_handleUpdateD
 	if applicationState.SyncError != "" {
 		condition.NewConditionManager().SetCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError, managedgitopsv1alpha1.GitOpsConditionStatusTrue, managedgitopsv1alpha1.GitopsDeploymentReasonSyncError, applicationState.SyncError)
 	} else {
-		// Update syncError Condition as false if applicationState.SyncError field in database is empty
-		reason := managedgitopsv1alpha1.GitopsDeploymentReasonSyncError + "Resolved"
-		if cond, _ := condition.NewConditionManager().FindCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError); cond.Reason != reason {
-			condition.NewConditionManager().SetCondition(&gitopsDeployment.Status.Conditions, managedgitopsv1alpha1.GitOpsDeploymentConditionSyncError, managedgitopsv1alpha1.GitOpsConditionStatusFalse, reason, "")
+		// Update syncError Condition as false if applicationState.SyncError field in database is empty by checking if the condition field is empty or not
```

Comment:
Rather than making the change for this story in `handleUpdateDeploymentStatusTick`, which would only be called once every 10 seconds, I think we should instead do the check (e.g. valid syntax check you wrote below) every time the user modifies the `GitOpsDeployment` resource (and thus we get a modification event for it). 

Whenever the user modified a `GitOpsDeployment` resource, the `applicationEventRunner_handleDeploymentModified` function will be called, so making the changes for this story in that function is the best bet.

The change can likely be made at the top of the function. Probably after `// 1) Retrieve the GitOpsDeployment from the namespace`.
- At this point, we can check if the GitOpsDeployment is valid, and if not, return an `error` if the path is blank. The error will be set as an `ErrorOccurred` condition.

Once this is done, I think the changes to `handleUpdateDeploymentStatusTick` can be removed.